File: manageEntrepriseAccounts/src/main/java/fr/polytech/unice/credirama/mea/entities/dto/TransactionRequest.java
package fr.polytech.unice.credirama.mea.entities.dto;

import fr.polytech.unice.credirama.mea.entities.TransactionType;
import lombok.*;

@Data
@AllArgsConstructor
@NoArgsConstructor
@ToString
@EqualsAndHashCode
public class TransactionRequest {

    private double amount;

    private TransactionType type;

    private int fromId;

    private int toId;

}

File: Flutter/credirama/lib/model/transactionRequest.dart
import 'dart:core';

class TransactionRequest {
  int fromId;
  int toId;
  double amount;
  static String type = "TRANSFER";

  TransactionRequest(this.fromId, this.toId, this.amount);

  Map<String, String> toSend() {
    Map<String, String> result = new Map();
    result.putIfAbsent("fromId", () => this.fromId.toString());
    result.putIfAbsent("toId", () => this.toId.toString());
    result.putIfAbsent("amount", () => this.amount.toString());
    result.putIfAbsent("type", () => type);
    return result;
  }

  @override
  String toString() {
    return 'TransactionRequest{fromId: $fromId, toId: $toId, amount: $amount}';
  }


}